text/docs-code-contributions.md

```diff
@@ -0,0 +1,69 @@
+- Start Date: 2019-02-07)
+- RFC PR: 
+- Gatsby Issue:
+
+# Summary
+
+Reorganize the contributor documentation to aid in Gatsby core and docs development.
+
+# Motivation
+
+The Gatsby.js docs, while wonderful, are light on coding instructions for contributors. With dedicated sections in the Contributing docs, it will be easier to learn how to clone the repository, where to find certain things like docs or plugins, how to run tests, and insider infrastructure knowledge required to submit a successful pull request on the first try.
+
+A lot of this information already lives on the "How to Contribute" page, but it's buried. It would help discoverability if this information lived on separate pages.
+
+# Detailed design
+
+I propose that we add some pages to the existing contributor documentation and move some content around. Here is an IA diagram that (briefly) reflects the current structure and what I think we should add (open for discussion, of course):
+
+Docs
+- Guides
+- Ecosystem
+- API Reference
+- Releases & Migration
+- Conceptual Guide
+- Behind the Scenes
+- Advanced Tutorials
+
+Contributing
+- Community
+    - Why contribute to Gatsby?
+    - Pair Programming Sessions
+    - Free swag for contributors
+    - Where to participate in the community
+    - How to run a Gatsby workshop*
+    - How to pitch Gatsby*
+- Code of Conduct
+- Gatsby Style Guide
+- How to contribute
+    * Filing & Managing issues
+    * Managing pull requests
+    * Setting up your local dev environment
+    * Docs contributions
+    * Code contributions
+    * Community contributions
+    * Showcase contributions
```

Comment:
I don't know if I'd group these as all "showcase contributions" because two of them are libraries, not showcases. Maybe "ecosystem contributions"? Not sure if the community links these things in their minds like I do!